service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/RemoveOrderControllerSubmittedEvent.java

```diff
@@ -0,0 +1,124 @@
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.order.CaseManagementOrder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(RemoveOrderController.class)
+@OverrideAutoConfiguration(enabled = true)
+class RemoveOrderControllerSubmittedEvent extends AbstractControllerTest {
+    @MockBean
+    private PopulateStandardDirectionsHandler populateStandardDirectionsHandler;
+
+    RemoveOrderControllerSubmittedEvent() {
+        super("remove-order");
+    }
+
+    @Test
+    void shouldPublishPopulateStandardDirectionsEventIfNewSDOHasBeenRemoved() {
+        StandardDirectionOrder previousSDO = StandardDirectionOrder.builder().build();
+
+        List<Element<StandardDirectionOrder>> hiddenSDOs = new ArrayList<>();
+        hiddenSDOs.add(element(previousSDO));
+
+        CaseDetails caseDetails = CaseDetails.builder().data(
+            Map.of("hiddenStandardDirectionOrders", hiddenSDOs)).build();
+
+        CaseDetails caseDetailsBefore = CaseDetails.builder().data(Map.of()).build();
+
+        CallbackRequest callbackRequest = CallbackRequest.builder()
+            .caseDetails(caseDetails)
+            .caseDetailsBefore(caseDetailsBefore)
+            .build();
+
+        postSubmittedEvent(callbackRequest);
+
+        verify(populateStandardDirectionsHandler).populateStandardDirections(any(
+            PopulateStandardDirectionsEvent.class));
+    }
+
+    @Test
+    void shouldPublishPopulateStandardDirectionsEventIfAnAdditionalSDOHasBeenRemoved() {
+        Element<StandardDirectionOrder> newSDO = element(StandardDirectionOrder.builder().build());
+        Element<StandardDirectionOrder> previousSDO = element(StandardDirectionOrder.builder().build());
+
+        List<Element<StandardDirectionOrder>> previousHiddenSDOs = new ArrayList<>();
+        previousHiddenSDOs.add(previousSDO);
+
+        List<Element<StandardDirectionOrder>> hiddenSDOs = new ArrayList<>();
+        hiddenSDOs.add(previousSDO);
+        hiddenSDOs.add(newSDO);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(Map.of(
+                "hiddenStandardDirectionOrders", hiddenSDOs
+            ))
+            .build();
+
+        CaseDetails caseDetailsBefore = CaseDetails.builder()
+            .data(Map.of(
+                "hiddenStandardDirectionOrders", previousHiddenSDOs
+            ))
+            .build();
+
+        CallbackRequest callbackRequest = CallbackRequest.builder()
+            .caseDetails(caseDetails)
+            .caseDetailsBefore(caseDetailsBefore)
+            .build();
+
+        postSubmittedEvent(callbackRequest);
+
+        verify(populateStandardDirectionsHandler).populateStandardDirections(any(
```

Comment:
using equalsTo rather than the any(class)?

Reply:
or can you assert on  coreCaseDataService.triggerEvent(caseDetails.getJurisdiction(),
            caseDetails.getCaseTypeId(),
            caseDetails.getId(),
            "populateSDO",
            caseDetails.getData()
        ); instead. Assuming that in the test is Sync